### ISSUE TITLE
overplotting in plotParameterSamples - Parameter Distribution 2D

### ISSUE THREAD

**Issue by FFroehlich** [ICB-DCM/PESTO#44]
plotting 2D parameter distributions with >10^5 samples (nsimu_run>10^6) takes long.
I think we should limit the maximum number of points to a reasonable number and use mvksdensity to color the plots by density

**Comment by BBallnus** [comment]
The reason why it takes so long is probably because each point is plotted separately and thus adding up the matlab figure overhead each time. Plotting all points at once speeds up the process a lot. Combining this with an plotting-internal thinning to a maximum point number should fix the problem.

Using density plots might be a nice option to scatter but I would not replace the "classic scatter" entirely since it provides a clearer view onto the non-processed results.

**Comment by FFroehlich** [comment]
I agree. But I did not suggest replacing the scatter plot, but instead to color the points of the scatterplot according to density (like this: http://mathematica.stackexchange.com/questions/85530/how-2d-scatterplots-with-quantitative-density-dependent-coloring). Would be an option to only activate this when a threshold of points is exceeded and the computed density is somewhat reasonable.

**Comment by paulstapor** [comment]
Holy crap, that's beautiful! :D

**Comment by FFroehlich** [comment]
the only matlab routine I know that does density scatters is https://de.mathworks.com/matlabcentral/fileexchange/8430-flow-cytometry-data-reader-and-visualization/content/dscatter.m but it is quite buggy. Yet, it really is only 2-3 lines of codes to implement this with mvksdensity (or better make an independent implementation that does not rely on the statistics toolbox).

**Comment by BBallnus** [comment]
fancy indeed!

**Comment by paulstapor** [comment]
I like the whole blueish thing [plot nr 2](http://mathematica.stackexchange.com/questions/85530/how-2d-scatterplots-with-quantitative-density-dependent-coloring), since it does process the data too much. The other pictures on the website are maybe a bit too much, although it might be an option thinking of some additional color coding for those regions which are completely filled with samples, to get even more information about the density. But here, things start to be very tricky...
